package/src/languageService/kustoLanguageService.ts

```diff
@@ -2031,7 +2031,8 @@ class KustoLanguageService implements LanguageService {
         // Replace new URL due to polyfill issue in IE
         // const hostname = new URL(schema.cluster.connectionString.split(';')[0]).hostname;
         const hostname = schema.cluster.connectionString.match(/(.*\/\/)?([^\/;]*)/)[2];
-        const clusterName = hostname.split('.')[0];
+        // safranke.eastus.kusto.windows.net --> safranke.eastus;
```

Comment:
Does this only apply to windows.net? If so, could we check for `.kusto.windows.net` to mitigate the odds we'll get a false positive?

Reply:
Are there any docs we could link to in a comment?

Reply:
AFAIK, checking '.kusto' make sense

* the cluster name when there is no '.kusto' (include else) doesn't seems right.
I would expect that if it not familiar name, the cluster name will be full host name or event full url (in log analytics) - 
   'kusto.aria.microsoft.com'/ 'something.playfab.com' or 'https://ade.applicationinsights.io/subscriptions/f3101802-8c4f-4e6e-819c-a3b5794d33dd/resourcegroups/kustoweb-europe/providers/microsoft.insights/components/kustoweb-ai'
  
* did you test .kustodev cluster and .kustomfa clusters ?

Reply:
Looks like the non kusto urls don't currently work with cross database query
I'll try to work with Matt to solve it somehow.
Merging this in the meantime.